internal/provider/resource_workspace.go

```diff
@@ -0,0 +1,330 @@
+	TerraformVersion   types.String `tfsdk:"terraform_version"`
+	PreventDestroyPlan types.Bool   `tfsdk:"prevent_destroy_plan"`
+	LastUpdated        types.String `tfsdk:"last_updated"`
+}
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &workspaceResource{}
+	_ resource.ResourceWithConfigure   = &workspaceResource{}
+	_ resource.ResourceWithImportState = &workspaceResource{}
+)
+
+// NewWorkspaceResource is a helper function to simplify the provider implementation.
+func NewWorkspaceResource() resource.Resource {
+	return &workspaceResource{}
+}
+
+type workspaceResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *workspaceResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_workspace"
+}
+
+// The diagnostics return value is required by the interface even though this function returns only nil.
+func (t *workspaceResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	description := "Defines and manages a workspace."
+
+	return tfsdk.Schema{
+		Version: 1,
+
+		MarkdownDescription: description,
+		Description:         description,
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:                types.StringType,
+				MarkdownDescription: "String identifier of the workspace.",
+				Description:         "String identifier of the workspace.",
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"name": {
+				Type:                types.StringType,
+				MarkdownDescription: "The name of the workspace.",
+				Description:         "The name of the workspace.",
+				Required:            true,
+			},
+			"description": {
+				Type:                types.StringType,
+				MarkdownDescription: "A description of the workspace.",
+				Description:         "A description of the workspace.",
+				Required:            true,
+			},
+			"full_path": {
+				Type:                types.StringType,
+				MarkdownDescription: "The path of the parent namespace plus the name of the workspace.",
+				Description:         "The path of the parent namespace plus the name of the workspace.",
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"group_path": {
+				Type:                types.StringType,
+				MarkdownDescription: "Path of the parent group.",
+				Description:         "Path of the parent group.",
+				Required:            true,
+			},
+			"max_job_duration": {
+				Type:                types.Int64Type,
+				MarkdownDescription: "Maximum job duration in minutes.",
+				Description:         "Maximum job duration in minutes.",
+				Optional:            true,
+			},
+			"terraform_version": {
```

Comment:
Seems like `computed` should be set to true on a few of these fields where the API will automatically set a default value if they are not provided.